stackslib/src/chainstate/nakamoto/signer_set.rs

```diff
@@ -487,4 +487,84 @@ impl NakamotoSigners {
         }
         Ok(signers)
     }
+
+    /// Verify that the transaction is a valid vote for the aggregate public key
+    /// Note: it does not verify the function arguments, only that the transaction is validly formed
+    /// and has a valid nonce from an expected address
+    pub fn valid_vote_transaction(
+        account_nonces: &HashMap<StacksAddress, u64>,
+        transaction: &StacksTransaction,
+        is_mainnet: bool,
+    ) -> bool {
+        let origin_address = transaction.origin_address();
+        let origin_nonce = transaction.get_origin_nonce();
+        let Some(account_nonce) = account_nonces.get(&origin_address) else {
+            debug!("valid_vote_transaction: Unrecognized origin address ({origin_address}).",);
+            return false;
+        };
+        if transaction.is_mainnet() != is_mainnet {
+            debug!("valid_vote_transaction: Received a transaction for an unexpected network.",);
+            return false;
+        }
+        if origin_nonce < *account_nonce {
+            debug!("valid_vote_transaction: Received a transaction with an outdated nonce ({account_nonce} < {origin_nonce}).");
+            return false;
+        }
+        Self::parse_vote_for_aggregate_public_key(transaction).is_some()
+    }
+
+    pub fn parse_vote_for_aggregate_public_key(
+        transaction: &StacksTransaction,
+    ) -> Option<(u64, Point, u64, u64)> {
```

Comment:
In general, I think large tuple function signatures should be avoided because they can be prone to errors (the type checker won't help you differentiate between one u64 and another), and I think its better to use one-off structs if necessary:

```suggestion
    pub struct AggregateKeyVoteParams {
        pub signer_index: u64,
        pub aggregate_key: Point,
        pub voting_round: u64,
        pub reward_cycle: u64,
    }
    pub fn parse_vote_for_aggregate_public_key(
        transaction: &StacksTransaction,
    ) -> Option<AggregateKeyVoteParams> {```

Reply:
Done :)